### PR TITLE
ci: add networking release validation

### PR DESCRIPTION
**Pull request by ethanndickson**
This calls a [networking performance regression test](https://github.com/coder/schmoder) whenever a release branch is cut. The results will be available via a Slack webhook in #netgru, and as a job artifact on the `schmoder` repo.

### CODE REVIEW

**Comment by ethanndickson** [comment]
* **#14273** <a href="https://app.graphite.dev/github/pr/coder/coder/14273?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/release-validation.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: release-validation
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  network-performance:
+    runs-on: ${{ github.repository_owner == 'coder' && 'depot-ubuntu-22.04-8' || 'ubuntu-latest' }}
```
Comment (ethanndickson) [inline .github/workflows/release-validation.yaml:10]: This is to please the CI linter.

Comment (matifali) [inline .github/workflows/release-validation.yaml:10]:
```suggestion
    runs-on: ubuntu-latest
```
As this action only triggers a repository dispatch event, it should not need an 8-core runner and would do well on the free `ubuntu-latest` runner.

I am assuming standard GitHub action linux runners are free to use on public repos i.e., `ubuntu-latest`
This [page](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) says:
> The use of these runners on public repositories is free and unlimited.

**Review by matifali — CHANGES_REQUESTED** [review]

File: .github/workflows/release-validation.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: release-validation
+
+on:
+  push:
+    tags:
+      - "v*"
+    branches: 08-15-ci_add_networking_release_validation
```
Comment (deansheather) [inline .github/workflows/release-validation.yaml:7]: remember to remove

**Review by deansheather — APPROVED** [review]

**Comment by ethanndickson** [comment]
Merging blocked by the requested changes.

**Review by matifali — APPROVED** [review]

**Pull request merged**